libfwupdplugin/fu-device.c

```diff
@@ -2906,6 +2916,16 @@ fu_device_detach (FuDevice *self, GError **error)
 	if (klass->detach == NULL)
 		return TRUE;
 
+	/* fixup old plugins */
+	if (priv->has_counterpart &&
+	    !fu_device_has_internal_flag (self, FU_DEVICE_INTERNAL_FLAG_REPLUG_MATCH_GUID)) {
+		g_warning ("device %s [%s] did not set FU_DEVICE_INTERNAL_FLAG_REPLUG_MATCH_GUID "
+			   "so setting it for backwards compat -- this will stop soon!",
+			   fu_device_get_name (self),
+			   fu_device_get_id (self));
+		fu_device_add_internal_flag (self, FU_DEVICE_INTERNAL_FLAG_REPLUG_MATCH_GUID);
```

Comment:
the number of plugins that use counterpart is a fixed number, why not just fix those immediately and set an error on detach?

Reply:
I think I fixed them all. There are actually plugins using counterpart for other reasons, e.g. the vli plugin uses it to find the USB-2 device that corresponds to the USB-3 device. Maybe dropping this chunk makes sense.